### DISCUSSION TITLE
Listening to events of Chrome DevTools

### DISCUSSION THREAD

**Discussion by sellem777** [php-webdriver/php-webdriver#880] [Q&A]
Hey,

How can I set an event listener for events fired by the chrome devtools, for example I would like to listen to the event Fetch.requestPaused and be able to respond to if from PHP.

Thanks

**Comment by OndraM** [comment]
I don't think this is possible directly via php-webdriver.

You can use [ChromeDevToolsDriver](https://github.com/php-webdriver/php-webdriver/wiki/Chrome#create-an-instance-of-chromedevtoolsdriver-class) to execute "raw" CDP commands. 

```
$driver = ChromeDriver::start();
$devTools = $driver->getDevTools();
$devTools->execute('Performance.enable');
$metrics = $devTools->execute('Performance.getMetrics');
```

But this just send "synchronous" request and receives response. Php-webdriver does not contain websocket-based connection to CDP, which would be needed for features like listeners.

As a full featured CDP protocol (but without Selenium features) you can use eg. https://github.com/jakubkulhan/chrome-devtools-protocol .
